### PR TITLE
gc: added --all-commits

### PR DESCRIPTION
**Pull request by Suor**
Plus fixed a bug in previous patch.

### CODE REVIEW

**Comment by efiop** [comment]
@Suor Tests failed.

**Comment by Suor** [comment]
Fixed it

File: dvc/command/gc.py [hunk 1]
```diff
@@ -14,8 +14,14 @@
 class CmdGC(CmdBase):
     def run(self):
         msg = "this will remove all cache except the cache that is used in "
-        if not self.args.all_branches and not self.args.all_tags:
+        if (
+            not self.args.all_branches
+            and not self.args.all_tags
+            and not self.args.all_commits
+        ):
             msg += "the current git branch"
```
Comment (efiop) [inline dvc/command/gc.py]: Btw, is it `current git branch` or `current workspace`?

File: tests/func/test_gc.py [hunk 2]
```diff
@@ -173,3 +175,37 @@ def test(self):
...
+    N = _count_files(cache_dir)
+
+    dvc_repo.gc(all_commits=True)
+
+    # Only one uncommited file should go away
+    assert _count_files(cache_dir) == N - 1
+
+
+def _count_files(path):
+    print(list(os.walk(path)))
```
Comment (efiop) [inline tests/func/test_gc.py]: is this needed?

**Comment by Suor** [comment]
Removed the print and fixed the warning. The last was actually a separate thing, so pretty commits here. Please merge @efiop ;)

**Comment by efiop** [comment]
> The last was actually a separate thing, so pretty commits here. Please merge @efiop ;)

@Suor Merges are disabled :D Only squash is available for consistency.

**Comment by Suor** [comment]
This was 3 unrelated commits.

**Comment by efiop** [comment]
@Suor I understand, but we now squash by default. If anyone needs more granularity, individual commits are available in the PR itself.

**Comment by efiop** [comment]
@Suor Please don't remove PR template. Did you submit a docs PR?

**Comment by efiop** [comment]
@Suor Created https://github.com/iterative/dvc.org/issues/730 . Please submit a doc PR.

**Comment by Suor** [comment]
@efiop this should not be documented.

**Pull request merged**